libsolidity/ast/AST.h

```diff
@@ -756,18 +756,41 @@ class TypeName: public ASTNode
 class ElementaryTypeName: public TypeName
 {
 public:
+	//For all types that do not have a length appended to them
 	ElementaryTypeName(SourceLocation const& _location, Token::Value _type):
 		TypeName(_location), m_type(_type)
 	{
 		solAssert(Token::isElementaryTypeName(_type), "");
 	}
+	//For uint<N>, int<N>, bytes<N>
+	ElementaryTypeName(SourceLocation const& _location, Token::Value _type, unsigned const& _N):
+		TypeName(_location), m_type(_type), m_N(_N)
+	{
+		solAssert(Token::isElementaryTypeName(_type), "");
+		annotation().hasN = true;
+		annotation().NxM = std::make_pair(m_N, 0);
```

Comment:
Is there a reason to duplicate N and M in the annotations? AST annotations are used to add information to an AST node after the parsing step. The N and M values can (and should) be determined at the stage where the node is constructed, so they should only be members here.
Could it be possible to have the following token types: `uint`, `int`, `uintM`, `intM`, `realMxN`, ..., so that we can see from the token type whether M or N is present?